CDCgov/prime-reportstream
# Added ability to send reports using PUT

This PR adds the ability to send reports using PUT through the REST transport and also sets the accessToken if provided in the credential.
Minnesota's REST API only accepts reports be sent to them using PUT. The `transport.headers.method` setting must be set to `PUT` so that the REST transport can configure the HTTP client to send via PUT, otherwise it will use POST by default.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Run `./prime multiple-settings set --input ./settings/STLTs/MN/mn-doh.yml`
2. Send test report routed to MN
3. Verify the report gets sent to MN
4. Query the `action` table by running `select * from action order by created_at desc`. Wait for an `action_name` with `send_error` entry to show up. The report fails because we're sending to a mock API that only validates the message and our messages fail that validation. The errors will be fixed, but for now this is only to test we can send reports to their REST API. We can verify that by looking at the `action_result` and making sure we get validation errors back.

## Changes
- Added ability to specify REST method inside transport.headers

## Checklist

### Testing

## Linked Issues
- Fixes #14696
